symengine/tests/basic/test_as_real_imag.cpp

```diff
@@ -259,3 +260,17 @@ TEST_CASE("RealImag: Trigonometric functions", "[as_real_imag]")
     CHECK_THROWS_AS(as_real_imag(asinh(i2), outArg(re), outArg(im)),
                     SymEngineException);
 }
+
+TEST_CASE("RealImag: Absolute Value Function", "[as_real_imag]")
+{
+    RCP<const Basic> re, im;
+    auto i2 = integer(2), i3 = integer(3);
+
+    as_real_imag(abs(add(add(i2, I), add(i3, I))), outArg(re), outArg(im));
+    REQUIRE(eq(*re, *sqrt(add(pow(add(i2, i3), i2), pow(i2, i2)))));
+    REQUIRE(eq(*im, *zero));
+
+    as_real_imag(abs(neg(i2)), outArg(re), outArg(im));
```

Comment:
These are numeric functions and `abs(-2)` is automatically simplified to `2`. Use a symbol inside `abs` like `abs(x-2)`